cluster/src/main/java/com/hartwig/pipeline/alignment/bwa/MergeMarkDups.java

```diff
@@ -1,32 +1,93 @@
 package com.hartwig.pipeline.alignment.bwa;
 
+import static java.lang.String.format;
+
 import java.util.List;
+import java.util.StringJoiner;
 
+import com.google.common.collect.Lists;
 import com.hartwig.pipeline.calling.command.SamtoolsCommand;
 import com.hartwig.pipeline.datatypes.FileTypes;
+import com.hartwig.pipeline.execution.vm.Bash;
 import com.hartwig.pipeline.execution.vm.BashCommand;
 import com.hartwig.pipeline.execution.vm.DeleteFilesCommand;
 import com.hartwig.pipeline.execution.vm.OutputFile;
+import com.hartwig.pipeline.execution.vm.SambambaCommand;
+import com.hartwig.pipeline.execution.vm.VmDirectories;
 import com.hartwig.pipeline.execution.vm.unix.RedirectStdoutCommand;
+import com.hartwig.pipeline.resource.ResourceFiles;
 import com.hartwig.pipeline.stages.SubStage;
 
 public class MergeMarkDups extends SubStage {
 
+    private final String sampleId;
+    private final ResourceFiles resourceFiles;
     private final List<String> inputBamPaths;
+    private final boolean useTargetRegions;
 
-    MergeMarkDups(final List<String> inputBamPaths) {
+    MergeMarkDups(
+            final String sampleId, final ResourceFiles resourceFiles, final List<String> inputBamPaths, boolean useTargetRegions) {
         super("", FileTypes.BAM);
+        this.sampleId = sampleId;
+        this.resourceFiles = resourceFiles;
         this.inputBamPaths = inputBamPaths;
+        this.useTargetRegions = useTargetRegions;
     }
 
     @Override
     public List<BashCommand> bash(final OutputFile input, final OutputFile output) {
+
         List<BashCommand> cmds = new java.util.ArrayList<>();
-        String intermediateOutputBAM = output.path() + ".intermediate.tmp";
-        cmds.add(new SambambaMarkdupCommand(inputBamPaths, intermediateOutputBAM));
-        cmds.add(new DeleteFilesCommand(inputBamPaths));
-        cmds.add(new RedirectStdoutCommand(new BamReheaderCommand(intermediateOutputBAM),  output.path()));
-        cmds.add(SamtoolsCommand.index(output.path()));
+
+        if(useTargetRegions) {
```

Comment:
The new MarkDups will soon (v5.34?) replace the Sambamba call - in this pipeline release it is only for targeted mode since that is where the main benefit will be.

Reply:
I have split the commands-list building into distinct methods